system-tests/azure-data-factory-tests/src/test/java/org/eclipse/dataspaceconnector/system/tests/local/AzureDataFactoryTransferIntegrationTest.java

```diff
@@ -0,0 +1,306 @@
+        if (!file.exists()) {
+            throw new FileNotFoundException("Runtime settings file not found");
+        }
+    }
+
+    @AfterAll
+    static void cleanUp() {
+        containerCleanup.parallelStream().forEach(Runnable::run);
+    }
+
+    @Test
+    public void transferBlob_success() {
+        // Arrange
+
+        // Upload a blob with test data on provider blob container
+        var providerBlobServiceClient = getBlobServiceClient(PROVIDER_STORAGE_ACCOUNT_NAME);
+        var consumerBlobServiceClient = getBlobServiceClient(CONSUMER_STORAGE_ACCOUNT_NAME);
+        var blobContent = "AzureDataFactoryTransferIntegrationTest-" + UUID.randomUUID();
+
+        providerBlobServiceClient
+                .createBlobContainer(PROVIDER_CONTAINER_NAME)
+                .getBlobClient(PROVIDER_ASSET_ID)
+                .upload(BinaryData.fromString(blobContent), true);
+        // Add for cleanup
+        containerCleanup.add(() -> providerBlobServiceClient.deleteBlobContainer(PROVIDER_CONTAINER_NAME));
+
+        // Seed data to provider
+        createAsset();
+        var policyId = createPolicy();
+        createContractDefinition(policyId);
+
+        // Act
+        System.setProperty(BlobTransferLocalSimulation.ACCOUNT_NAME_PROPERTY, CONSUMER_STORAGE_ACCOUNT_NAME);
+        runGatling(BlobTransferLocalSimulation.class, TransferSimulationUtils.DESCRIPTION);
+
+        // Assert
+        var provisionedContainerName = getProvisionedContainerName();
+        // Add for cleanup
+        containerCleanup.add(() -> consumerBlobServiceClient.deleteBlobContainer(provisionedContainerName));
+
+        var destinationBlob = consumerBlobServiceClient.getBlobContainerClient(provisionedContainerName)
+                .getBlobClient(PROVIDER_ASSET_ID);
+        assertThat(destinationBlob.exists())
+                .withFailMessage("Destination blob %s not created", destinationBlob)
+                .isTrue();
+        var actualBlobContent = destinationBlob.downloadContent().toString();
+        assertThat(actualBlobContent)
+                .withFailMessage("Transferred file contents are not same as the source file")
+                .isEqualTo(blobContent);
+
+    }
+
+    @NotNull
+    private static BlobServiceClient getBlobServiceClient(String accountName) {
+        var credential = new DefaultAzureCredentialBuilder().build();
+        var vault = new SecretClientBuilder()
+                .vaultUrl(format(KEY_VAULT_ENDPOINT_TEMPLATE, KEY_VAULT_NAME))
+                .credential(credential)
+                .buildClient();
+        var accountKey = vault.getSecret(accountName + "-key1");
+        return new BlobServiceClientBuilder()
+                .endpoint(format(BLOB_STORE_ENDPOINT_TEMPLATE, accountName))
+                .credential(new StorageSharedKeyCredential(accountName, accountKey.getValue()))
+                .buildClient();
+    }
+
+    @NotNull
+    private String getProvisionedContainerName() {
```

Comment:
how about creating `azure-tests` `testFixtures` to avoid the duplication of a number of methods here?